geonode/layers/templates/layers/layer_detail.html

```diff
@@ -293,6 +293,7 @@ <h4 class="modal-title" id="myModalLabel">{% trans "Download Layer" %}</h4>
                               {% for link in links_download %}
                               <li><a href="{{ link.url }}" target="_blank">{% trans link.name %}</a></li>
                               {% endfor %}
+                              <li><a href="{% url 'qgis_server:download-qlr' layername=resource.name %} " target="_blank">QGIS Layer file</a></li>
```

Comment:
@boney-bun You should add the QLR as a link in the DB for this layer, no? Then, it will be like L294.

And this link is only for QGIS Server, but here, you add it for all layers inlcuding geoserver.